### PR TITLE
Paulius/files&folders integration tests

### PR DESCRIPTION
**Pull request by pgrigaliunas**
Added and splitted integration tests for ProjectShareClient

### CODE REVIEW

File: clients/projectshare/src/test/ProjectShareClient.test.ts [hunk 1]
```diff
@@ -154,7 +425,10 @@ describe("ProjectShareClient (#integration)", () => {
   }
 
   it("should be able to CRUD custom properties", async () => {
-    const firstImageFile = (await projectShareClient.getFiles(requestContext, projectId, new ProjectShareFileQuery().startsWithPathAndNameLike(projectId, "360-Images/2A", "2A_v0410470")))[0];
+    let FoldersInRootFolder: ProjectShareFolder[] = await projectShareClient.getFolders(requestContext, projectId, new ProjectShareFolderQuery().inRootFolder(projectId));
+    await CreateFoldersWithFiles(projectShareClient, requestContext, projectId, FoldersInRootFolder[0].wsgId)
```
Comment (tomaspakala) [inline clients/projectshare/src/test/ProjectShareClient.test.ts:429]: Is it really worth it to use CreateFoldersWithFiles in every test?

**Review by tomaspakala — COMMENTED** [review]
Reviewing integration tests changes

File: clients/projectshare/src/test/ProjectShareClient.test.ts (hunk 1 above)
Comment (pgrigaliunas) [reply clients/projectshare/src/test/ProjectShareClient.test.ts:429]: in this test CreateFoldersWithFiles is necessary

File: clients/projectshare/src/test/ProjectShareClient.test.ts [hunk 2]
```diff
@@ -24,128 +77,346 @@ describe("ProjectShareClient (#integration)", () => {
...
+    await CreateFoldersWithFiles(projectShareClient, requestContext, projectId, FoldersInRootFolder[0].wsgId);
 
-    // inRootFolder
+    // act
     let files: ProjectShareFile[] = await projectShareClient.getFiles(requestContext, projectId, new ProjectShareFileQuery().inRootFolder(projectId));
+
+    // assert
     chai.assert(files);
-    chai.assert.strictEqual(0, files.length);
+    chai.assert.strictEqual(files.length, 0);
```
[the remark this answers is not in the document]
Comment (pgrigaliunas) [reply clients/projectshare/src/test/ProjectShareClient.test.ts:253]: Yes

File: common/api/presentation-components.api.md [hunk 3]
```diff
@@ -165,10 +165,10 @@ export interface DataProvidersFactoryProps {
 export const DEFAULT_PROPERTY_GRID_RULESET: Ruleset;
 
 // @beta @deprecated
-export function DEPRECATED_controlledTreeWithFilteringSupport<P extends ControlledTreeWithVisibleNodesProps>(TreeComponent: React.FC<P>): React.FC<Pick<P & ControlledTreeWithFilteringSupportProps, "filter" | "onFilterApplied" | "onMatchesCounted" | "activeMatchIndex" | "nodeLoader" | "onNodeLoaderChanged" | Exclude<keyof P, "visibleNodes">>>;
+export function DEPRECATED_controlledTreeWithFilteringSupport<P extends ControlledTreeWithVisibleNodesProps>(TreeComponent: React.FC<P>): React.FC<Pick<P & ControlledTreeWithFilteringSupportProps, "filter" | "nodeLoader" | "onFilterApplied" | "onMatchesCounted" | "onNodeLoaderChanged" | "activeMatchIndex" | Exclude<keyof P, "visibleNodes">>>;
```
Comment (grigasp) [inline common/api/presentation-components.api.md:168]: will need to
```
rush clean
rush compile
rush extract-api
```
to clean up these unintentional changes

**Review by grigasp — CHANGES_REQUESTED** [review]

**Comment by calebmshafer** [comment]
@pgrigaliunas there are a handful of lint errors in the project-share package.  Can you run "npm run lint" in that package and take a look?

**Comment by pgrigaliunas** [comment]
> @pgrigaliunas there are a handful of lint errors in the project-share package. Can you run "npm run lint" in that package and take a look?

Fixed

**Comment by calebmshafer** [comment]
@Mergifyio backport release/2.10.0

**Pull request merged**